### ISSUE TITLE
Fuzzer failure in map_keys(array_frequency(array_sort("c0")))

### ISSUE THREAD

**Issue by aditi-pandit** [facebookincubator/velox#3912]
### Description

Fuzzer failed with error on expression map_keys(array_frequency(array_sort("c0")))

Different results at idx '1': '[1->19] 9 elements starting at 28 {gTH@%l-@8wXM}XlG, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, 97oN-2v"<As<67!i$lK8kY%%%3E_s#+U=,NdMlbR!^2Yq|^&F"\w5k!svsO3U9DaWB(@a6"JMN-YZtdLYdXQ~heN, ?d%Y<+y>pU$, >`nMTniA~4lf]obhGS<[i~b"(:SpcBylyc3-qT2?\;\W91oX&PKw{6CgqoO, ...}' vs. '9 elements starting at 0 {Wo^d`1m_-t"F?]dnfJxK1w129ChD<u=@D>)</8CNH%#U7Pkrww]VLS+z$VfD5a9wYwI, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, k#=qr|\,}r39Hr, -&4hC?T9p/fh52V.i+urrbP1pC&)+{qY6\yRvc^?Us)qk+<j'GSfV~U'=xD3lP\]`Wkt-p`okj{X4ECO%+i)hZr<WC, gTH@%l-@8wXM}XlG, ...}', 

### Relevant logs

```bash
I0202 01:14:06.288791 143056 ExpressionVerifier.cpp:80] Executing expression: map_keys(array_frequency(array_sort("c0")))
I0202 01:14:06.288815 143056 ExpressionVerifier.cpp:32] 1 vectors as input:
I0202 01:14:06.288822 143056 ExpressionVerifier.cpp:34] 	[DICTIONARY ARRAY<VARCHAR>: 100 elements, 10 nulls], [DICTIONARY ARRAY<VARCHAR>: 100 elements, 12 nulls], [DICTIONARY ARRAY<VARCHAR>: 100 elements, 8 nulls], [DICTIONARY ARRAY<VARCHAR>: 100 elements, 13 nulls], [ARRAY ARRAY<VARCHAR>: 100 elements, no nulls]
E0202 01:14:06.290728 143056 Exceptions.h:68] Line: ../../velox/expression/tests/ExpressionVerifier.cpp:62, Function:compareVectors, Expression: left->equalValueAt(right.get(), i, i) Different results at idx '1': '[1->19] 9 elements starting at 28 {gTH@%l-@8wXM}XlG, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, 97oN-2v"<As<67!i$lK8kY%%%3E_s#+U=,NdMlbR!^2Yq|^&F"\w5k!svsO3U9DaWB(@a6"JMN-YZtdLYdXQ~heN, ?d%Y<+y>pU$, >`nMTniA~4lf]obhGS<[i~b"(:SpcBylyc3-qT2?\;\W91oX&PKw{6CgqoO, ...}' vs. '9 elements starting at 0 {Wo^d`1m_-t"F?]dnfJxK1w129ChD<u=@D>)</8CNH%#U7Pkrww]VLS+z$VfD5a9wYwI, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, k#=qr|\,}r39Hr, -&4hC?T9p/fh52V.i+urrbP1pC&)+{qY6\yRvc^?Us)qk+<j'GSfV~U'=xD3lP\]`Wkt-p`okj{X4ECO%+i)hZr<WC, gTH@%l-@8wXM}XlG, ...}', Source: RUNTIME, ErrorCode: INVALID_STATE
I0202 01:14:06.291555 143056 ExpressionVerifier.cpp:295] Persisted input: --fuzzer_repro_path /tmp/fuzzer_repro/velox_expressionVerifier_TJmvPc --input_path /tmp/fuzzer_repro/velox_expressionVerifier_TJmvPc/input_vector --sql_path /tmp/fuzzer_repro/velox_expressionVerifier_TJmvPc/sql
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Different results at idx '1': '[1->19] 9 elements starting at 28 {gTH@%l-@8wXM}XlG, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, 97oN-2v"<As<67!i$lK8kY%%%3E_s#+U=,NdMlbR!^2Yq|^&F"\w5k!svsO3U9DaWB(@a6"JMN-YZtdLYdXQ~heN, ?d%Y<+y>pU$, >`nMTniA~4lf]obhGS<[i~b"(:SpcBylyc3-qT2?\;\W91oX&PKw{6CgqoO, ...}' vs. '9 elements starting at 0 {Wo^d`1m_-t"F?]dnfJxK1w129ChD<u=@D>)</8CNH%#U7Pkrww]VLS+z$VfD5a9wYwI, vrmIT(+^"OF!(7g?HAM=/\QD'9{1'#_rvw]St%8nk+R3n7*]x(GR=<efB0Z)7qRH]l, k#=qr|\,}r39Hr, -&4hC?T9p/fh52V.i+urrbP1pC&)+{qY6\yRvc^?Us)qk+<j'GSfV~U'=xD3lP\]`Wkt-p`okj{X4ECO%+i)hZr<WC, gTH@%l-@8wXM}XlG, ...}'
Retriable: False
Expression: left->equalValueAt(right.get(), i, i)
Function: compareVectors
File: ../../velox/expression/tests/ExpressionVerifier.cpp
Line: 62
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxException5State4makeIZNS1_C4EPKcmS5_St17basic_string_viewIcSt11char_traitsIcEES9_S9_S9_bNS1_4TypeES9_EUlRT_E_EESt10shared_ptrIKS2_ESA_SB_
# 2  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 3  _ZN8facebook5velox17VeloxRuntimeErrorC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bS7_
# 4  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 5  _ZN8facebook5velox4test12_GLOBAL__N_114compareVectorsERKSt10shared_ptrINS0_10BaseVectorEES7_
# 6  _ZN8facebook5velox4test18ExpressionVerifier6verifyERKSt10shared_ptrIKNS0_4core10ITypedExprEERKS3_INS0_9RowVectorEEOS3_INS0_10BaseVectorEEbSt6vectorIjSaIjEE
# 7  _ZN8facebook5velox4test16ExpressionFuzzer2goEv
# 8  _ZN8facebook5velox4test16expressionFuzzerESt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt6vectorIPKNS0_4exec17FunctionSignatureESaISD_EESt4hashIS8_ESt8equal_toIS8_ESaISt4pairIKS8_SF_EEEm
# 9  _ZN12FuzzerRunner3runERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmRKSt13unordered_setIS5_St4hashIS5_ESt8equal_toIS5_ESaIS5_EES7_
# 10 main
# 11 __libc_start_main
# 12 _start

*** Aborted at 1675300446 (Unix time, try 'date -d @1675300446') ***
*** Signal 6 (SIGABRT) (0x22ed0) received by PID 143056 (pthread TID 0x7fa7d82dd300) (linux TID 143056) (maybe from PID 143056, UID 0) (code: -6), stack trace: ***
(error retrieving stack trace)
/bin/bash: line 13: 143056 Aborted                 (core dumped) _build/debug/velox/expression/tests/velox_expression_fuzzer_test --seed 123456 --enable_variadic_signatures --velox_fuzzer_enable_complex_types --lazy_vector_generation_ratio 0.2 --velox_fuzzer_enable_column_reuse --velox_fuzzer_enable_expression_reuse --duration_sec 60 --logtostderr=1 --minloglevel=0 --repro_persist_path=/tmp/fuzzer_repro

Exited with code exit status 134
CircleCI received exit code 134
```

**Comment by kagamiori** [comment]
This failure is due to the non-determinism of the order of keys in folly::F14FastMap used in the implementation of array_frequency().

**Comment by pedroerp** [comment]
This is related to the other Issue I claimed (#3045); I can take a look at this one as well.

**Comment by kagamiori** [comment]
> This is related to the other Issue I claimed (#3045); I can take a look at this one as well.

Hi Pedro, just FYI, array_frequency() has been added to our skip list a month ago for the reason of non-determinism. So this won't fail in CircleCI or at diff time. It would still be nice if we can have a way to test it and remove it from the skip list.

**Comment by pedroerp** [comment]
This has been fixed in #4435
